### PR TITLE
Less allocated objects on each request

### PR DESCRIPTION
**Pull request by u2**
http://www.sitepoint.com/unraveling-string-key-performance-ruby-2-2/
https://github.com/rack/rack/pull/737

```ruby
module V1
  module App
    class Ping < Grape::API
      format :json
      get '/ping' do
        present :ping, :pong
      end
    end
  end
end
```

```ruby
require 'rails_helper'
require 'memory_profiler'

describe ::API::API do
  describe "GET /v1/api/app/ping.json" do
    it "return 200" do
      report = MemoryProfiler.report do
        get "/v1/api/app/ping.json"
      end

      report.pretty_print
      expect(response.status).to eq(200)
    end
  end
end
```
before:
```grape_upstream/lib x 10369233```

after:
```grape/lib x 10366633```

```(10369233 - 10366633)/10369233.to_f * 100  #=> 0.02507417858196455```% fewer objects per request.

Maybe there are some missed optimizations.I will updated later.

### CODE REVIEW

File: lib/grape.rb [hunk 1]
```diff
@@ -27,6 +27,17 @@
 require 'i18n'
 require 'thread'
 
+CONTENT_TYPE = 'Content-Type'.freeze
```
Comment (dblock) [inline lib/grape.rb]: I don't like how these are globals. They are all HTTP headers, so create a module, like `Grape::HTTP::Headers` and stick them in there?

**Comment by u2** [comment]
I am not sure the `Grape::HTTP::Headers` suitable or not.

**Comment by u2** [comment]
Why `eql Hash.new` passed, and `eql {}` failed?
I think `Hash.new.eql?({}) #=> true`.

**Comment by dm1try** [comment]
> and eql {} failed?
 because here `{}` was interpreted as a block, use parenthesesю

**Comment by u2** [comment]
Yes, thank you. @dm1try

File: lib/grape.rb [hunk 2]
```diff
@@ -26,6 +26,7 @@
 require 'virtus'
 require 'i18n'
 require 'thread'
+require 'grape/http/headers'
```
Comment (dblock) [inline lib/grape.rb]: Why is this required?

File: lib/grape.rb [hunk 3]
```diff
@@ -133,6 +134,8 @@ module Presenters
   end
 end
 
+include Grape::Http::Headers
```
Comment (dblock) [inline lib/grape.rb]: Maybe group with the above `require`, but in theory this shouldn't even be necessary since this class should be found anyway, no?

Comment (u2) [inline lib/grape.rb]: Not be necessary.I just want to write `HTTP_ACCEPT_VERSION` instead of `Grape::Http::Headers::HTTP_ACCEPT_VERSION`, less words,but maybe it's not a good idea.

**Comment by u2** [comment]
Could you close this PR, I create a another one #977
